web/src/runtime.ts

```diff
@@ -1477,6 +1489,41 @@ export class Instance implements Disposable {
     this.cacheMetadata = { ...this.cacheMetadata, ...(list["metadata"] as Record<string, any>) };
   }
 
+  /**
+   * Delete NDArray from cache url.
+   *
+   * @param ndarrayCacheUrl
+   * @param device
+   * @param cacheScope
+   */
+  async deleteNDArrayCache(
+    ndarrayCacheUrl: string,
+    device: DLDevice,
+    cacheScope = "tvmjs"
+  ): Promise<any>{
+    const artifactCache = new ArtifactCache(cacheScope);
+    const jsonUrl = new URL("ndarray-cache.json", ndarrayCacheUrl).href;
+    const result = await artifactCache.fetchWithCache(jsonUrl);
+    let list;
+    if (result instanceof Response){
+      list = await result.json();
+    }
+    const arrayentry = list["records"] as Array<NDArrayShardEntry>;
+    for (let i = 0; i < arrayentry.length; ++i){
+      const dataUrl = new URL(arrayentry[i].dataPath, ndarrayCacheUrl).href;
+      try {
+        const result = await(await artifactCache.deleteInCache(dataUrl));
+        if (result === false){
+          return false;
+        }
```

Comment:
Should we end the deletion if one of the entry is not found? Or should we ensure everything inside `ndarry-cache` is deleted? I prefer the latter -- let me know what you think! If we go with the latter, this function does not need to return a `boolean` but can be `void`. Whichever we go with, we need to make sure the function's behavior is well defined, matching its docstring.

Reply:
I also prefer the latter one. Will change the function to void